tests/testTriggers.go

```diff
@@ -813,6 +815,132 @@ func TriggerDeployNewApps(contexts *[]*scheduler.Context, recordChan *chan *Even
+				dash.VerifyFatal(err==nil, true, "Find storage cluster ?")
+				var nodeId string
+				storageNodes:= node.GetStorageNodes()
+				nodeId = storageNodes[0].VolDriverNodeID
+				err = Inst().N.DetachDrivesFromVM(stc, storageNodes[0].Name)
+				UpdateOutcome(event, err)
+				time.Sleep(time.Duration(1 * time.Minute))
+				statusErr := Inst().V.WaitDriverUpOnNode(storageNodes[0], 10*time.Minute)
+				if statusErr != nil {
+					if strings.Contains(statusErr.Error(), apios.Status_STATUS_STORAGE_DOWN.String()) {
+						log.Infof("Node has gone to Storage Down state after detach %v: Node %s", statusErr, storageNodes[0].Name)
+					}else{
+						UpdateOutcome(event, statusErr)
+					}
+				}
+				storageNode, err := Inst().V.GetDriverNode(&storageNodes[0])
+				UpdateOutcome(event, err)
+				if storageNode.Status.String() != apios.Status_STATUS_STORAGE_DOWN.String(){
+					UpdateOutcome(event, fmt.Errorf("Node %s: Expected: %v Actual: %v", storageNode.SchedulerNodeName,
+					apios.Status_STATUS_STORAGE_DOWN, storageNode.Status))
+				}
+				log.Infof("Status of the storage node %s ,%v ", storageNode.SchedulerNodeName, storageNode.Status)
+				statusErr = Inst().V.EnterMaintenance(storageNodes[0])
+				UpdateOutcome(event, statusErr)
+				status, _ := Inst().V.GetNodeStatus(storageNodes[0])
+				log.Infof("Status when the storage node entered maintenance mode %s , ", status.String())
+				if status.String() != apios.Status_STATUS_MAINTENANCE.String(){
+					UpdateOutcome(event, fmt.Errorf("Node %s: Expected: %v Actual: %v", storageNodes[0].Name,
+					apios.Status_STATUS_MAINTENANCE, status))
+				}
+				statusErr = Inst().V.ExitMaintenance(storageNodes[0])
+				UpdateOutcome(event, statusErr)
+				status, _ = Inst().V.GetNodeStatus(storageNodes[0])
+				log.Infof("Node Status after exit maintenance mode %v , %s", status, storageNodes[0].Name)
+				if status.String() != apios.Status_STATUS_OK.String(){
+					UpdateOutcome(event, fmt.Errorf("Node %s: Expected: %v Actual: %v", storageNodes[0].Name,
+					apios.Status_STATUS_MAINTENANCE, status))
+				}
+				err = Inst().V.RefreshDriverEndpoints()
+				UpdateOutcome(event, err)
+				//There are chances that storageless node can convert storage node by attaching these drives. Hence making sure
+				//This node id is up.
+				var selectedStorageNode *node.Node
+				storageNodes = node.GetStorageNodes()
+				for _, nodeInfo := range storageNodes {
+					if nodeInfo.VolDriverNodeID == nodeId {
+						selectedStorageNode = &nodeInfo
+						log.Infof("Node which has the nodeID %v , %s", status, (*selectedStorageNode).Name)
+						break
+					}
+				}
+				if selectedStorageNode != nil{
+					status, err = Inst().V.GetNodeStatus(*selectedStorageNode)
+					if err != nil {
+						UpdateOutcome(event, err)
+					}else{
+						log.Infof("Node Status  %v node: %s", status, (*selectedStorageNode).SchedulerNodeName)
+						if status.String() != apios.Status_STATUS_OK.String(){
```

Comment:
if node id is same doest it mean that same driveset is added?

Reply:
Yes. node id is nothing but driveset id, if storageless node becomes storagenode then nodeId is moved to new storage node.